src/python/pants/goal/exec_stats.py

```diff
@@ -0,0 +1,137 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import base64
+import logging
+import textwrap
+from collections import Counter
+from io import BytesIO
+from typing import cast
+
+from pants.engine.internals.scheduler import Workunit
+from pants.engine.rules import collect_rules, rule
+from pants.engine.streaming_workunit_handler import (
+    StreamingWorkunitContext,
+    WorkunitsCallback,
+    WorkunitsCallbackFactory,
+    WorkunitsCallbackFactoryRequest,
+)
+from pants.engine.unions import UnionRule
+from pants.option.subsystem import Subsystem
+
+logger = logging.getLogger(__name__)
+
+
+class ExecStatsSubsystem(Subsystem):
+    options_scope = "exec-stats"
+    help = "An aggregator for Pants execution stats, such as cache metrics."
+
+    @classmethod
+    def register_options(cls, register):
+        register(
+            "--log",
```

Comment:
Bike shed #2, what should this be called? Some options:

* `--log`
* `--show`
* `--on`
* `--enable`